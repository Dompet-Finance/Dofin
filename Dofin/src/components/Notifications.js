'use strict'

import React from 'react'

import {
  View,
  Text,
  Button,
} from 'react-native'

class AppBar extends React.Component {
  render() {
    return (
      <View>
        <Text>
          AppBar
        </Text>
      </View>
    )
  }
}

class NotificationList extends React.Component {
  render() {
    return (
      <View>
        <Text>
          Notification List
        </Text>
      </View>
    )
  }
}

export class Transactions extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      income: 0,
      expense: 0,
    }
  }

  render() {
    return (
      <View>
        <AppBar />
        <NotificationList />
        { this.state.expense > this.state.income
          &&
          (<Text>
            Overlimit
          </Text>)
        }
      </View>
    )
  }
}

const styles = {
}

export default Transactions