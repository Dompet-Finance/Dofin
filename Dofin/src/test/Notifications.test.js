'use strict'

import 'react-native'
import 'react-dom'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import Notifications from '../components/Notifications.js'

describe('Test Notifications component', () => {

  let wrapper
  beforeEach(() => {
    wrapper = shallow(<Notifications />)
  })

  test('renders correctly', () => {
    const tree = renderer.create(
      <Notifications />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  })

  test('should render appbar', () => {
    const child = wrapper.find('AppBar')
    expect(child).toHaveLength(1)
  })

  test('should render notification list', () => {
    const child = wrapper.find('NotificationList')
    expect(child).toHaveLength(1)
  })

})

