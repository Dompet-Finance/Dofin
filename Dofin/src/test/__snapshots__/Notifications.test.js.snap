// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Notifications component renders correctly 1`] = `
<View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      AppBar
    </Text>
  </View>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Notification List
    </Text>
  </View>
</View>
`;
